permission/acl/acl.go

```diff
@@ -45,3 +46,25 @@ func ValidRawAccount(accountName string) error {
 	}
 	return nil
 }
+
+func validContractName(contractName string) error {
+	// param absence check
+	// contract naming rule check
+	contractSize := len(contractName)
+	contractMaxSize := utils.GetContractNameMaxSize()
+	contractMinSize := utils.GetContractNameMinSize()
+	if contractSize > contractMaxSize || contractSize < contractMinSize {
+		return fmt.Errorf("contract name length expect [%d~%d], actual: %d", contractMinSize, contractMaxSize, contractSize)
+	}
+	if m, _ := regexp.MatchString("^[a-z,A-Z,_]{1}[0-9,a-z,A-Z,_,.]{14}[0-9,a-z,A-Z,_]{1}$", contractName); !m {
+		return fmt.Errorf("contract name does not fit the rule of contract name")
+	}
+	return nil
+}
+
+// ValidContractName check if contract name is ok
+// rules: contract name has the length of 16 including a-z,A-Z,0-9,_.
+// rules: the character of a-z,A-Z,_ is accepted.
```

Comment:
two rules not the same?

Reply:
I have deleted the comments in the latest commit.